### DISCUSSION TITLE
Rebuild time is 8 times longer than initial build

### DISCUSSION THREAD

**Discussion by ikwihad** [webpack/webpack#17381] [Q&A]
delete

**Comment by ikwihad** [comment]
deleted

**Reply by alexander-akait** [reply]
Please provide reproducible test repo or profile and put a file here

**Comment by ikwihad** [comment]
I started making a repository and probably figured it out myself. It was  that I ran the compilation of the server and client bundles in the same process, and this resulted in a build build time that was higher than the starting volume build. I launched first `compiler.watch(server.config)` and then launched `new WebpackDevServer(client.config.devServer)`, but then I changed compiler.watch to compiler.run and everything worked as it should. perhaps they cannot work in parallel, and the second build can only be built after the first

**Reply by alexander-akait** [reply]
@ikwihad Do you use `webpack-cli`? If no - why? Also we have `dependecies` and webpack itself will run build in the right order

**Reply by ikwihad** [reply]
delete

**Comment by ikwihad** [comment]
delete

**Comment by ikwihad** [comment]
I came to this architecture because my server is restarted with nodemon, the client with webpack. And I don't want time to be wasted on re-creating the front-end parts when the server is restarted

maybe there is a better option?

**Reply by alexander-akait** [reply]
Sorry, your architecture is exotic for me, and I can't help you here without documentation and examples, I don't want to say, it is wrong/bad, it just takes time to learn and undestand

**Comment by ikwihad** [comment]
what is the standard architecture for applications with server and client rendering? do maybe have example repo with this?

**Reply by alexander-akait** [reply]
@ikwihad There is not standard :smile: developers use different approaches, I just want to say that in order to help here I will need to understand your architecture, but unfortunately I'm not sure I can find the time for this, so if you can formulate short and understandable questions, I can help with them
